src/devices/src/virtio/block/io/async_io.rs

```diff
@@ -27,7 +27,7 @@ pub enum Error {
 
 pub struct AsyncFileEngine<T> {
     file: File,
-    ring: IoUring,
+    ring: IoUring<WrappedUserData<T>>,
     completion_evt: EventFd,
     phantom: PhantomData<T>,
```

Comment:
Do we need this any more?

Reply:
Yes. I moved generic to IoUring struct it self. It helps removing unsafe code.

Reply:
I meant that I think we can remove the Phantom now